Fix DIND to make the images more reusable

ivan4th opened the pull request:
1. Add a comment about `STOPSIGNAL SIGRTMIN+3` in xenial-systemd-bare.dkr
2. Force Docker to use VFS driver. It'll default to devicemapper otherwise which doesn't always work & may cause problems with loop device exhaustion
3. Add dind script based on Docker's own dind setup. Added `mount --make-shared /` there because otherwise there may be problems with volume mounts [according](https://github.com/marun/dkr-systemd-dind/blob/25702cf7a4a73bcc355a492f4e5ca96cc562a5a5/dind-setup.sh#L46) to @marun

Commit:
Add a comment on what STOPSIGNAL SIGRTMIN+3 means

Commit:
Implement proper DIND setup

Based on ideas from Docker's own DIND image
and https://github.com/kubernetes/release/pull/86#issuecomment-248185616

errordeveloper commented on images/build-debian.sh:
```diff
@@ -52,4 +52,7 @@ apt-get -qq -y install socat
 apt-get -qq clean
 
+# Use vfs driver for docker
+sed -i 's/\(ExecStart=.*\)/\1 --storage-driver=vfs/' /lib/systemd/system/docker.service
```
Do you mind to add a drop-in instead? [See the docs.](https://docs.docker.com/engine/admin/systemd/)

ivan4th commented:
oops, amended wrong commit, will fix it in a moment

ivan4th commented:
Fixed.

errordeveloper commented on images/xenial-systemd-base.dkr:
```diff
@@ -1,3 +1,6 @@
+ADD ./dind /usr/local/bin/dind
 RUN /tmp/build.sh
+
+ENTRYPOINT ["/usr/local/bin/dind"]
```
IS it suppose to wrapp `/sbin/init`?

errordeveloper commented:
I'm curious if the entrypoint script can be implemented as a chain of systemd units... Looks good overall, but please clarify one comment.

errordeveloper commented:
Would you mind to address @marun's https://github.com/kubernetes/release/pull/86#issuecomment-248185616 as well?

ivan4th commented:
I've updated the PR. Incorporated @marun's suggestions except dangling volume part (it needs to be handled outside the image & only when the volumes are actually created). Also made it execute dind setup as a systemd unit. I used a shell script that's invoked from .service file for that, could use some .mount files maybe but I think this may complicate things a bit.
